core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java

```diff
@@ -20,45 +20,65 @@
  * #L%
  */
 
+import com.google.common.base.Throwables;
+
 import org.apache.hadoop.security.UserGroupInformation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.security.PrivilegedExceptionAction;
 import java.sql.Connection;
 import java.sql.SQLException;
+import java.util.concurrent.Callable;
 
+import javax.annotation.Nonnull;
 import javax.sql.DataSource;
 
 /**
+ * Utility methods for executing code that requires an active Kerberos ticket.
  */
 public class KerberosUtil {
 
     private static final Logger log = LoggerFactory.getLogger(KerberosUtil.class);
 
+    /**
+     * Gets a connection for the specified data source using the optional Kerberos ticket.
+     */
     public static Connection getConnectionWithOrWithoutKerberos(final DataSource dataSource, KerberosTicketConfiguration kerberosTicketConfiguration) throws SQLException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.sql.SQLException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)